lib/Sema/SemaBounds.cpp

```diff
@@ -520,22 +518,77 @@ namespace {
             assert("unexpected cast failure");
             return CreateBoundsNone();
           }
-          switch (UO->getOpcode()) {
-            case UnaryOperatorKind::UO_AddrOf: {
-              Expr *SubExpr = UO->getSubExpr();
-              if (SubExpr->getType()->isFunctionType())
-                return CreateBoundsNone();
+          UnaryOperatorKind Op = UO->getOpcode();
 
-              return LValueBounds(SubExpr);
-            }
-            default:
-              // TODO: fill in other cases
+          if (Op == UnaryOperatorKind::UO_AddrOf) {
+            Expr *SubExpr = UO->getSubExpr();
+            if (SubExpr->getType()->isFunctionType())
               return CreateBoundsNone();
+
+              return LValueBounds(SubExpr);
           }
+
+          if (UnaryOperator::isIncrementDecrementOp(Op))
+            return LValueTargetBounds(UO->getSubExpr());
+
+          if (Op == UnaryOperatorKind::UO_Plus ||
+              Op == UnaryOperatorKind::UO_Minus ||
+              Op == UnaryOperatorKind::UO_Not)
+            return RValueBounds(UO->getSubExpr());
+          return CreateBoundsNone();
         }
-        // TODO: fill in these cases
         case Expr::BinaryOperatorClass:
-        case Expr::CompoundAssignOperatorClass:
+        case Expr::CompoundAssignOperatorClass: {
+          BinaryOperator *BO = dyn_cast<BinaryOperator>(E);
+          if (!BO) {
+            assert("unexpected cast failure");
```

Comment:
`assert("<string>")` will never trigger, as strings are truthy. You probably want `assert(false && "<string>")`. I should have caught this several pull-requests ago.